tests/MongoDB.Driver.Tests/AggregateGraphLookupEnumerableFromOrToTests.cs

```diff
@@ -0,0 +1,241 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Collections.Generic;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using Xunit;
+
+namespace MongoDB.Driver.Tests
+{
+    public class AggregateGraphLookupTests
+    {
+        // public methods
+        [SkippableFact]
+        public void GraphLookup_with_many_to_one_parameters_should_return_expected_result()
+        {
+            RequireServer.Check().Supports(Feature.AggregateGraphLookupStage);
+            var database = GetDatabase();
+            var collectionName = "collectionManyToOne";
+            EnsureTestDataManyToOne(database, collectionName);
+            var expectedResult = new ManyToOneResult[]
+            {
+                new ManyToOneResult
+                {
+                    Id = 1,
+                    From = new int[] { 2, 3 },
+                    To = 1,
+                    Matches = new List<ManyToOne> { new ManyToOne { Id = 2, From = new[] { 3, 4 }, To = 2 } }
+                },
+                new ManyToOneResult
+                {
+                    Id = 2,
+                    From = new[] { 3, 4 },
+                    To = 2,
+                    Matches = new List<ManyToOne>()
+                }
+            };
+            var collection = database.GetCollection<ManyToOne>(collectionName);
+
+            var result = collection
+                .Aggregate()
+                .GraphLookup(
+                    from: collection,
+                    connectFromField: x => x.From,
+                    connectToField: x => x.To,
+                    startWith: x => x.From,
+                    @as: (ManyToOneResult x) => x.Matches)
+                .ToList();
+
+            result.Count.Should().Be(2);
+            result[0].ToBsonDocument().Should().Be(expectedResult[0].ToBsonDocument());
+            result[1].ToBsonDocument().Should().Be(expectedResult[1].ToBsonDocument());
+        }
+
+        [SkippableFact]
+        public void GraphLookup_with_one_to_many_parameters_should_return_expected_result()
+        {
+            RequireServer.Check().Supports(Feature.AggregateGraphLookupStage);
+            var database = GetDatabase();
+            var collectionName = "collectionOneToMany";
+            EnsureTestDataOneToMany(database, collectionName);
```

Comment:
Ditto above

Reply:
I feel like separate private methods are quite readable, so I leave it as is.